overlord/hookstate/ctlcmd/mount.go

```diff
@@ -126,6 +145,11 @@ func (m *mountCommand) matchConnection(attributes map[string]interface{}) bool {
 		return false
 	}
 
+	// TODO we do exact match on the mount options, which means that plugs
+	// referencing filesystems, which may require authentication options passed
+	// in -o <option-list>, would need to spell out all authentication options
+	// directly in plug declaration. This may be unacceptable in certain
+	// scenarios, e.g. CIFS with user=foo,password=foo options.
```

Comment:
I suppose the issue is more the extra testing that would be needed because the change here itself would seem simple